lib/HLSL/HLMatrixLowerPass.cpp

```diff
@@ -537,6 +545,24 @@ void HLMatrixLowerPass::replaceAllVariableUses(
       continue;
     }
 
+    if (BitCastInst *BCI = dyn_cast<BitCastInst>(Use.getUser())) {
+      // Replace bitcasts to i8* for lifetime intrinsics.
+      if (BCI->getType()->isPointerTy() && BCI->getType()->getPointerElementType()->isIntegerTy(8))
+      {
+        DXASSERT(BCI->getNumUses() == 1, "expected only a single use");
+        llvm::Use &UseUse = *BCI->use_begin();
+        IntrinsicInst *Intrin = dyn_cast<IntrinsicInst>(UseUse.getUser());
+        DXASSERT(Intrin && (Intrin->getIntrinsicID() == Intrinsic::lifetime_start || Intrin->getIntrinsicID() == Intrinsic::lifetime_end), "use must be lifetime.start/end");
```

Comment:
The simplest way to fix the Clang builds would be to use `DXASSERT_LOCALVAR` here and provide Intrin as the first variable. You could also move the `dyn_cast<IntrinsicInst>(UseUse.getUser())` into the DXASSERT.

Reply:
Fixed.
I couldn't repro the failure, though, is there anything special I have to do? building the ALL_BUILD target in VS in Release mode did not trigger it, at least not in an incremental build.

Reply:
The failures were in Linux and Apple builds using the clang compiler, which has warnings as errors that don't perfectly overlap with those produced by MSVC, but can help reveal programming mistakes that apply to all platforms.

Reply:
I see, thanks!